tests/framework/cli/test_starters.py

```diff
@@ -85,6 +85,42 @@ def test_starter_list(fake_kedro_cli):
         assert alias in result.output
 
 
+def test_starter_list_with_starter_plugin(fake_kedro_cli, entry_point):
+    """Check that `kedro starter list` prints out the plugin starters."""
+    entry_point.load.return_value = [KedroStarterSpec("valid_starter", "valid_path")]
+    entry_point.module = "valid_starter_module"
+    result = CliRunner().invoke(fake_kedro_cli, ["starter", "list"])
+    assert result.exit_code == 0, result.output
+    assert "valid_starter_module" in result.output
+
+
+@pytest.mark.parametrize(
+    "specs,expected",
+    [
+        (
+            [{"alias": "valid_starter", "template_path": "valid_path"}],
+            "should be a 'KedroStarterSpec'",
+        ),
+        (
+            [
+                KedroStarterSpec("duplicate", "duplicate"),
+                KedroStarterSpec("duplicate", "duplicate"),
+            ],
+            "has been ignored as it is already defined by",
+        ),
+    ],
+)
+def test_starter_list_with_invalid_starter_plugin(
+    fake_kedro_cli, entry_point, specs, expected
+):
+    """Check that `kedro starter list` prints out the plugin starters."""
+    entry_point.load.return_value = specs
+    entry_point.module = "invalid_starter"
+    result = CliRunner().invoke(fake_kedro_cli, ["starter", "list"])
+    assert result.exit_code == 0, result.output
+    assert expected in result.output
+
+
```

Comment:
do you intend to test the user prompt?

Reply:
The tests below in `TestNewFromUserPromptsValid` and `TestNewFromUserPromptsInvalid` check the user prompts, or did you mean something else?

Reply:
@MerelTheisenQB  Sorry forget about it.  I was looking [here](https://github.com/kedro-org/kedro/pull/1643/files/0620c5ba06fe17bccdfa8e34a91fe9e6b030cc0e..8e85558b1187c7a85fdec479584758ae63e668e1#) and saw changes like `test_starter_list_with_starter_plugin`. I think I am just confused by the merging changes. It's not the changes made in this PR.